dandiapi/api/views/dandiset.py

```diff
@@ -538,3 +598,37 @@ def clear_uploads(self, request, dandiset__pk):
 
         dandiset.uploads.all().delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @swagger_auto_schema(
+        methods=['POST'],
+        manual_parameters=[DANDISET_PK_PARAM],
+        request_body=no_body,
+        responses={
+            200: 'Dandiset starred successfully',
+            401: 'Authentication required',
+        },
+        operation_summary='Star a dandiset.',
+        operation_description='Star a dandiset. User must be authenticated.',
+    )
+    @action(methods=['POST'], detail=True)
+    def star(self, request, dandiset__pk) -> Response:
+        dandiset = self.get_object()
+        star_count = star_dandiset(user=request.user, dandiset=dandiset)
+        return Response({'count': star_count}, status=status.HTTP_200_OK)
+
+    @swagger_auto_schema(
+        methods=['POST'],
+        manual_parameters=[DANDISET_PK_PARAM],
+        request_body=no_body,
+        responses={
+            200: 'Dandiset unstarred successfully',
+            401: 'Authentication required',
+        },
+        operation_summary='Unstar a dandiset.',
+        operation_description='Unstar a dandiset. User must be authenticated.',
+    )
+    @action(methods=['POST'], detail=True)
+    def unstar(self, request, dandiset__pk) -> Response:
+        dandiset = self.get_object()
+        star_count = unstar_dandiset(user=request.user, dandiset=dandiset)
+        return Response({'count': star_count}, status=status.HTTP_200_OK)
```

Comment:
We could unify these two functions into one if we had them as
- `POST /dandiset/{pk}/star`
- `DELETE /dandiset/{pk}/star`
(you'd just have a conditional on the method to select between `star_dandiset` and `unstar_dandiset` and the rest of the function is identical between the two) and it would be more idiomatically "RESTful".

But this existing design is (RESTfully) valid as well. @mvandenburgh, @jjnesbitt, what do you think?

Reply:
I actually like the POST/DELETE better, that seems more inline with the RESTful philosophy of POST==create and DELETE==destroy

Reply:
I agree, good idea. I'll make that change.

Reply:
Changed in [afe79a3](https://github.com/dandi/dandi-archive/pull/2123/commits/afe79a35ef57a2c59cea5ec804529d389b358076)